### PR TITLE
General improvements

### PR DESCRIPTION
**Pull request by dominikl**
Some general improvements to the R gateway:

- Added a `getGroupContext` and `setGroupContext` method to allow switching group context.
- Also added optional `group` parameter to the `connect` method.
- Made the `port` parameter optional (if not specified uses default 4064).
- Avoid some unnecessary print outs of returned object (e. g. when calling `disconnect`).

**Based on #45 !!**

# Testing
Here's some sample code which logs in (default group) and get the available datasets; then change group to 'read-annotate-1' and then get the available datasets for that group again (e.g use user-3 (default group should be 'read-only-1')).
```
server <- OMEROServer(host = "eel.openmicroscopy.org", username = "xxx", password = "xxx")
server <- connect(server)

datasets <- getDatasets(server)
length(datasets)
getGroupContext(server)  # should print out 'read-only-1'

server <- setGroupContext(server, "read-annotate-1")
datasets <- getDatasets(server)
length(datasets)    # should be different from the above number
```

### CODE REVIEW

**Comment by dominikl** [comment]
Looks good now. Would be great to get it in for `0.4.0` !

File: tests/testthat/test-Image.R [hunk 1]
```diff
@@ -18,3 +18,21 @@ test_that("Test Image getThumbnail",{
   expect_that(dim(thumb), equals(c(96,96,3)))
 })
 
+
+test_that("Test Image getPixelValues",{
+  img <- loadObject(server, "ImageData", imageID)
+
+  pixels <- getPixelValues(img, 1, 1, 1)
```
Comment (jburel) [inline tests/testthat/test-Image.R]: have you checked value with invalid value i.e. 0, negative, value > sizeZ?

Comment (jburel) [reply tests/testthat/test-Image.R]: I don't think so.
Maybe declare that an exception will be thrown if the specified values are out of range
so users know what to handle

**Comment by dominikl** [comment]
I should actually add an integration test for the new methods. But could be done in a follow up PR, too.

**Comment by jburel** [comment]
Could you add them today before we tag? 
You can add to this PR

**Comment by jburel** [comment]
Before 7ea972b

```
OK:       158
Failed:   0
Warnings: 0
Skipped:  0
```
After
```
Duration: 20.7 s

OK:       159
Failed:   0
Warnings: 0
Skipped:  0
```

**Comment by jburel** [comment]
Looks good. Merging and tagging

**Pull request merged**